### PR TITLE
Sage Modal

### PR DESCRIPTION
**Pull request by AndrwM**
## Description
- Sage Modal
- Removed docs-specific button styles because it was messing with the layout of buttons nested within the modal
- This will be revisited for refinement once implemented in Build's Admin Navigation Management project (next week)
- @croswell has given this a +1

### Screenshots
image

## Related
#287

### CODE REVIEW

**Review by pixelflips — APPROVED** [review]
👍 +1 LGTM - Nice work and a great initial set up.

File: app/views/examples/objects/modal/_preview.html.erb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<div class="sage-btn sage-btn--primary sage-btn--icon-right-launch" data-js-modal-trigger="cool-modal">Trigger Modal</div>
```
Comment (teenwolfblitzer) [inline app/views/examples/objects/modal/_preview.html.erb]: Getting extremely nitpicky here, but could we make this into a `button`? 😁

File: app/views/examples/objects/modal/_markup.html.erb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<div class="sage-modal" data-js-modal="cool-modal" role="dialog">
```
Comment (teenwolfblitzer) [inline app/views/examples/objects/modal/_markup.html.erb]: Minor item: could we also add in `aria-modal="true"` as an attribute? In theory it should help screenreaders remove focus from the background.

File: lib/sage-frontend/stylesheets/system/patterns/objects/_modal.scss [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+$-modal-padding-x: sage-spacing(lg);
+$-modal-padding-y: sage-spacing(md);
+
+.sage-modal {
+  display: flex;
+  visibility: hidden;
+  align-items: flex-start;
+  justify-content: center;
+  position: fixed;
+  z-index: (sage-z-index(modal) - 1);
```
Comment (teenwolfblitzer) [inline lib/sage-frontend/stylesheets/system/patterns/objects/_modal.scss]: You may be able to use `sage-z-index(modal, -1)` here, though I haven't tested the negative increment

**Review by teenwolfblitzer — APPROVED** [review]
👍 🔥 Great work! A few minor suggestions, nothing blocking.

**Comment by AndrwM** [comment]
@monkeypox8 – Landing with the nits fixed above 💯

**Pull request merged**